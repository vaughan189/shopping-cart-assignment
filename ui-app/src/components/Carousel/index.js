import React from "react";
import Carousel from "react-material-ui-carousel";
import { CardActionArea } from "@mui/material";
import Card from "@mui/material/Card";
import CardMedia from "@mui/material/CardMedia";

import "./index.scss";

export const Carousal = (props) => {
  const { bannerData } = props;

  return (
    <Carousel className="CarouselItem">
      {bannerData.map((item, i) => (
        <CarouselItem
          key={i}
          item={item}
          bannerImageAlt={item.bannerImageAlt}
          bannerImageUrl={item.bannerImageUrl}
        />
      ))}
    </Carousel>
  );
};

export const CarouselItem = (props) => {
  const { bannerImageUrl, bannerImageAlt } = props;
  return (
    <Card>
      <CardActionArea>
        <CardMedia
          component="img"
          height="400"
          image={bannerImageUrl}
          alt={bannerImageAlt}
        />
      </CardActionArea>
    </Card>
  );
};
