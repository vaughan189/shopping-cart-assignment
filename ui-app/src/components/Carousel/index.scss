.CarouselItem {
  width: 100%;
  margin-top: 20px;
}

img {
  max-width: 100%;
  height: auto;
  padding: 0;
  margin: 0;
}
