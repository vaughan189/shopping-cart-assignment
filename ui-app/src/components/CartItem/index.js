import React from "react";

import {
  Card,
  CardActionArea,
  CardContent,
  CardMedia,
  IconButton,
  Typography,
} from "@mui/material";

import AddSharpIcon from "@mui/icons-material/AddSharp";
import RemoveSharpIcon from "@mui/icons-material/RemoveSharp";
import { Box } from "@mui/system";

export const CartItem = (props) => {
  const { cartItem, deleteCartItem, addProductsToCart } = props;
  return (
    <Card sx={{ margin: 1 }}>
      <CardActionArea>
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-between",
          }}
        >
          <CardMedia
            sx={{
              width: { xl: 100, md: 50, sm: 100, xs: 100 },
              height: { xl: 100, md: 50, sm: 100, xs: 100 },
            }}
            component="img"
            height="20"
            image={cartItem.imageURL}
            alt="green iguana"
          />
          <CardContent
            sx={{ width: { xl: "100%", md: "100%", sm: "100%", xs: "100%" } }}
          >
            <Box
              sx={{
                display: "flex",
                flexDirection: "column",
              }}
            >
              <Typography
                gutterBottom
                variant="h6"
                component="div"
                sx={{
                  fontWeight: "Bold",
                  fontSize: {
                    xl: "1rem",
                    md: "1rem",
                    sm: "1rem",
                    xs: "1rem",
                  },
                }}
              >
                {cartItem.productName}
              </Typography>
              <Box
                sx={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "space-between",
                  alignItems: "center",
                }}
              >
                <Box
                  sx={{
                    display: "flex",
                    flexDirection: "row",
                    justifyContent: "space-evenly",
                    alignItems: "flex-start",
                    width: { xl: "40%", md: "40%", sm: "40%", xs: "70%" },
                  }}
                >
                  <IconButton
                    id="subtract-Cart-Item"
                    aria-label="subtract"
                    size="small"
                    sx={{
                      height: "20px",
                      width: "20px",
                      borderRadius: { xl: 5, md: 5, sm: 0, xs: 0 },
                      color: "#FFFFFF",
                      backgroundColor: "#c73f6d",
                      "&.MuiButtonBase-root:hover": {
                        bgcolor: "#c73f6d",
                      },
                    }}
                    onClick={() => deleteCartItem(cartItem.id)}
                  >
                    <RemoveSharpIcon />
                  </IconButton>
                  <Typography
                    sx={{ width: "10px", fontFamily: "Dosis" }}
                    component="div"
                  >
                    {cartItem.quantity}
                  </Typography>
                  <IconButton
                    id="add-Cart-Item"
                    aria-label="add"
                    size="small"
                    sx={{
                      height: "20px",
                      width: "20px",
                      borderRadius: { xl: 5, md: 5, sm: 0, xs: 0 },
                      color: "#FFFFFF",
                      backgroundColor: "#c73f6d",
                      "&.MuiButtonBase-root:hover": {
                        bgcolor: "#c73f6d",
                      },
                    }}
                    onClick={() => addProductsToCart(cartItem)}
                  >
                    <AddSharpIcon />
                  </IconButton>
                  <Typography component="div" sx={{ fontFamily: "Dosis" }}>
                    X
                  </Typography>
                  <Typography component="div" sx={{ fontFamily: "Dosis" }}>
                    Rs.{cartItem.price}
                  </Typography>
                </Box>
                <Box
                  sx={{
                    display: "flex",
                    flexDirection: "row",
                    justifyContent: "space-around",
                    alignItems: "center",
                  }}
                >
                  <Typography component="div" sx={{ fontFamily: "Dosis" }}>
                    Rs. {cartItem.price * cartItem.quantity}
                  </Typography>
                </Box>
              </Box>
            </Box>
          </CardContent>
        </Box>
      </CardActionArea>
    </Card>
  );
};
