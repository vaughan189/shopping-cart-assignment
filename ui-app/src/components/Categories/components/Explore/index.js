import React from "react";
import { Button, Grid } from "@mui/material";
import { useHistory } from "react-router-dom";

import "./index.scss";

export const Explore = (props) => {
  const { name, description } = props;
  let history = useHistory();

  return (
    <Grid
      container
      direction="column"
      justifyContent="center"
      alignItems="center"
      spacing={3}
    >
      <Grid item xs={12} sm={12} md={12} xl={12} className="categoryName">
        {name}
      </Grid>
      <Grid item xs={12} sm={12} md={12} xl={12}>
        {description}
      </Grid>
      <Grid item xs={12} sm={12} md={12} xl={12}>
        <Button
          id="explore-more-btn"
          variant="contained"
          color="primary"
          style={{ textTransform: "none" }}
          onClick={() => history.push("/products")}
        >
          Explore {name}
        </Button>
      </Grid>
    </Grid>
  );
};
