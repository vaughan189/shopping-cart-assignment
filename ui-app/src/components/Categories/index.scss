.CategoriesItem {
  margin-top: 10px;
}

img {
  max-width: 100%;
  height: auto;
  padding: 0;
  margin: 0;
}
