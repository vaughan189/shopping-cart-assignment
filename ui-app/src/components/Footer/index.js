import { Box, Typography } from "@mui/material";
import React from "react";

export const Footer = () => {
  return (
    <Typography
      sx={{
        color: "#000000",
        position: "fixed",
        bottom: " 0px",
        width: "100%",
        height: "5%",
        backgroundColor: "#bebdbd",
      }}
    >
      <Box
        sx={{
          marginLeft: "16%",
          padding: "16px",
        }}
      >
        <Typography>
          Copyright Â© 201-2018 Sabka Bazaar Grocery Supplies Pvt Ltd
        </Typography>
      </Box>
    </Typography>
  );
};
