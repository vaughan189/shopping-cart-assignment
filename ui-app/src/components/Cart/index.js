/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from "react";

import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import { CartItem } from "../CartItem";
import { Button, IconButton } from "@mui/material";
import { LowestPriceSection } from "../LowestPrice";

import ArrowForwardIosSharpIcon from "@mui/icons-material/ArrowForwardIosSharp";
import useCart from "../../hooks/useCart";

export const Cart = () => {
  const { cartProductsData, cartProductList, cartProductsDataLoader } =
    useCart();
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "space-between",
        fontFamily: "Dosis",
      }}
    >
      <CartHeader />
      {!cartProductsDataLoader && (
        <>
          {cartProductList.map((item) => {
            return (
              <Box key={item.id}>
                <CartItem cartItem={item} />
              </Box>
            );
          })}
        </>
      )}
      <LowestPriceSection />
      <CheckOutSection cartProductsData={cartProductsData} />
    </Box>
  );
};

const CartHeader = () => {
  return (
    <Typography
      component="div"
      sx={{
        fontSize: 18,
        height: "40px",
        paddingTop: 2,
        paddingLeft: 2,
        fontWeight: "bold",
        backgroundColor: {
          xl: "#000000",
          md: "#000000",
          sm: "#FFFFFF",
          xs: "#FFFFFF",
        },
        color: {
          xl: "#FFFFFF",
          md: "#FFFFFF",
          sm: "#000000",
          xs: "#000000",
        },
        fontFamily: "Dosis",
      }}
    >
      My Cart{" "}
      <Typography
        component="span"
        sx={{
          fontFamily: "Dosis",
          fontSize: 16,
        }}
      >
        (1 item)
      </Typography>
    </Typography>
  );
};

const CheckOutSection = (props) => {
  const { cartProductsData } = props;
  const [totalPrice, setTotalPrice] = useState(0);

  const getTotalPrice = () => {
    return cartProductsData.data.reduce(
      (a, b) => {
        return {
          price: a.price + b.price,
        };
      },
      {
        price: 0,
      }
    );
  };

  useEffect(() => {
    if (
      cartProductsData &&
      cartProductsData.data &&
      cartProductsData.data.length > 0
    ) {
      setTotalPrice(getTotalPrice());
    }
  }, [cartProductsData]);

  return (
    <Box
      sx={{
        bottom: 0,
        display: "flex",
        flexDirection: "column",
        justifyContent: "centre",
        alignItems: "centre",
        marginTop: { xl: "70%", md: "70%", sm: "88%", xs: "83%" },
        backgroundColor: "white",
        padding: 2,
      }}
    >
      <Typography
        sx={{
          display: "flex",
          flexDirection: "row",
          justifyContent: "center",
          fontFamily: "Dosis",
          fontSize: 16,
          fontWeight: 600,
        }}
      >
        Promo code can be applied on payment page
      </Typography>
      <Button
        variant="contained"
        sx={{
          display: "flex",
          flexDirection: "row",
          justifyContent: "space-between",
          alignItems: "center",
        }}
      >
        <Typography component="span">Proceed to Checkout</Typography>
        <Box
          sx={{
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-around",
            alignItems: "center",
          }}
        >
          <Typography component="span" sx={{ textTransform: "none" }}>
            Rs.{totalPrice.price}
          </Typography>
          <IconButton aria-label="delete" size="small" color="secondary">
            <ArrowForwardIosSharpIcon
              fontSize="inherit"
              sx={{ color: "#FFFFFF" }}
            />
          </IconButton>
        </Box>
      </Button>
    </Box>
  );
};
