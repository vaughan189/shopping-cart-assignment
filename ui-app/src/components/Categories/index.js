import React from "react";
import { Card, CardMedia, Grid } from "@mui/material";

import "./index.scss";
import { Explore } from "./components/Explore";

export const Categories = (props) => {
  const { categoriesData } = props;
  return (
    <>
      {categoriesData.map((item, index) => {
        return (
          <CategoriesItem
            key={item.id}
            name={item.name}
            description={item.description}
            imageUrl={item.imageUrl}
            position={index % 2}
          />
        );
      })}
    </>
  );
};

export const CategoriesItem = (props) => {
  const { name, description, imageUrl, position } = props;
  return (
    <Card className="CategoriesItem" sx={{ minHeight: 260 }}>
      <Grid
        container
        direction="row"
        justifyContent="center"
        alignItems="center"
      >
        {position === 0 && (
          <>
            <Grid item xs={4} sm={4} md={4} xl={4}>
              <CardMedia
                rel="preload"
                component="img"
                height="400"
                image={imageUrl}
                alt={name}
              />
            </Grid>
            <Grid item xs={6} sm={6} md={6} xl={6}>
              <Explore name={name} description={description} />
            </Grid>
          </>
        )}
        {position === 1 && (
          <>
            <Grid item xs={6} sm={6} md={6} xl={6}>
              <Explore name={name} description={description} />
            </Grid>
            <Grid item xs={4} sm={4} md={4} xl={4}>
              <CardMedia
                rel="preload"
                component="img"
                height="400"
                image={imageUrl}
                alt={name}
              />
            </Grid>
          </>
        )}
      </Grid>
    </Card>
  );
};
