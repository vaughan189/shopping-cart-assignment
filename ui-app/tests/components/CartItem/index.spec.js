import React from "react";
import { shallow } from "enzyme";
import { CartItem } from "../../../src/components/CartItem";

describe("CartItem component", () => {
  it("should render the CartItem component correctly", () => {
    const props = {
      cartItem: {
        productId: "5b6c6a7f01a7c38429530883",
        productName: "Fresho Kiwi - Green, 3 pcs",
        imageURL: "/static/images/products/fruit-n-veg/kiwi-green.jpg",
        price: 87,
        quantity: 1,
        id: 1,
      },
      deleteCartItem: jest.fn(),
      addProductsToCart: jest.fn(),
    };
    const component = shallow(<CartItem {...props} debug />);
    expect(component).toMatchSnapshot();
  });

  it("should trigger the subtract item function on clicking the button", () => {
    const props = {
      cartItem: {
        productId: "5b6c6a7f01a7c38429530883",
        productName: "Fresho Kiwi - Green, 3 pcs",
        imageURL: "/static/images/products/fruit-n-veg/kiwi-green.jpg",
        price: 87,
        quantity: 1,
        id: 1,
      },
      deleteCartItem: jest.fn(),
      addProductsToCart: jest.fn(),
    };
    const component = shallow(<CartItem {...props} debug />);
    component.find('#subtract-Cart-Item').simulate('click');
    expect(props.deleteCartItem).toHaveBeenCalled()
  });

  it("should trigger the add item function on clicking the button", () => {
    const props = {
      cartItem: {
        productId: "5b6c6a7f01a7c38429530883",
        productName: "Fresho Kiwi - Green, 3 pcs",
        imageURL: "/static/images/products/fruit-n-veg/kiwi-green.jpg",
        price: 87,
        quantity: 1,
        id: 1,
      },
      deleteCartItem: jest.fn(),
      addProductsToCart: jest.fn(),
    };
    const component = shallow(<CartItem {...props} debug />);
    component.find('#add-Cart-Item').simulate('click');
    expect(props.addProductsToCart).toHaveBeenCalled()
  });
});
