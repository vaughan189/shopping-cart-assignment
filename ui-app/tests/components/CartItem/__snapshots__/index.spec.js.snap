// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component should render the CartItem component correctly 1`] = `
<ForwardRef(Card)
  sx={
    Object {
      "margin": 1,
    }
  }
>
  <ForwardRef(CardActionArea)>
    <ForwardRef(Box)
      sx={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <ForwardRef(CardMedia)
        alt="green iguana"
        component="img"
        height="20"
        image="/static/images/products/fruit-n-veg/kiwi-green.jpg"
        sx={
          Object {
            "height": Object {
              "md": 50,
              "sm": 100,
              "xl": 100,
              "xs": 100,
            },
            "width": Object {
              "md": 50,
              "sm": 100,
              "xl": 100,
              "xs": 100,
            },
          }
        }
      />
      <ForwardRef(CardContent)
        sx={
          Object {
            "width": Object {
              "md": "100%",
              "sm": "100%",
              "xl": "100%",
              "xs": "100%",
            },
          }
        }
      >
        <ForwardRef(Box)
          sx={
            Object {
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <ForwardRef(Typography)
            component="div"
            gutterBottom={true}
            sx={
              Object {
                "fontSize": Object {
                  "md": "1rem",
                  "sm": "1rem",
                  "xl": "1rem",
                  "xs": "1rem",
                },
                "fontWeight": "Bold",
              }
            }
            variant="h6"
          >
            Fresho Kiwi - Green, 3 pcs
          </ForwardRef(Typography)>
          <ForwardRef(Box)
            sx={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <ForwardRef(Box)
              sx={
                Object {
                  "alignItems": "flex-start",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                  "width": Object {
                    "md": "40%",
                    "sm": "40%",
                    "xl": "40%",
                    "xs": "70%",
                  },
                }
              }
            >
              <ForwardRef(IconButton)
                aria-label="subtract"
                id="subtract-Cart-Item"
                onClick={[Function]}
                size="small"
                sx={
                  Object {
                    "&.MuiButtonBase-root:hover": Object {
                      "bgcolor": "#c73f6d",
                    },
                    "backgroundColor": "#c73f6d",
                    "borderRadius": Object {
                      "md": 5,
                      "sm": 0,
                      "xl": 5,
                      "xs": 0,
                    },
                    "color": "#FFFFFF",
                    "height": "20px",
                    "width": "20px",
                  }
                }
              >
                <Memo(ForwardRef(RemoveSharpIcon)) />
              </ForwardRef(IconButton)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "fontFamily": "Dosis",
                    "width": "10px",
                  }
                }
              >
                1
              </ForwardRef(Typography)>
              <ForwardRef(IconButton)
                aria-label="add"
                id="add-Cart-Item"
                onClick={[Function]}
                size="small"
                sx={
                  Object {
                    "&.MuiButtonBase-root:hover": Object {
                      "bgcolor": "#c73f6d",
                    },
                    "backgroundColor": "#c73f6d",
                    "borderRadius": Object {
                      "md": 5,
                      "sm": 0,
                      "xl": 5,
                      "xs": 0,
                    },
                    "color": "#FFFFFF",
                    "height": "20px",
                    "width": "20px",
                  }
                }
              >
                <Memo(ForwardRef(AddSharpIcon)) />
              </ForwardRef(IconButton)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "fontFamily": "Dosis",
                  }
                }
              >
                X
              </ForwardRef(Typography)>
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "fontFamily": "Dosis",
                  }
                }
              >
                Rs.
                87
              </ForwardRef(Typography)>
            </ForwardRef(Box)>
            <ForwardRef(Box)
              sx={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <ForwardRef(Typography)
                component="div"
                sx={
                  Object {
                    "fontFamily": "Dosis",
                  }
                }
              >
                Rs. 
                87
              </ForwardRef(Typography)>
            </ForwardRef(Box)>
          </ForwardRef(Box)>
        </ForwardRef(Box)>
      </ForwardRef(CardContent)>
    </ForwardRef(Box)>
  </ForwardRef(CardActionArea)>
</ForwardRef(Card)>
`;
